src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -23,18 +23,17 @@ private static int GetConfig(string configName, int defaultValue)
                 switch (config)
                 {
                     case string str:
-                        int result;
                         if (str.StartsWith("0x"))
                         {
-                            if(!int.TryParse(str.Substring(2), NumberStyles.HexNumber, NumberFormatInfo.InvariantInfo, out result))
-                            {
-                                return defaultValue;
-                            }
-                            return result;
+                            return Convert.ToInt32(str, 16);
```

Comment:
Also, hexadecimal and octal parsing should only support unsigned values. It would be useful to have helpers to parse certain types (bool, int, uint should do for now). Only int/uint would need to support hex/octal, and those should use Convert.ToUInt32, followed by a range check and cast to int if it's in range.